v0/destinations/tiktok_ads/transform.js

```diff
@@ -0,0 +1,261 @@
+/* eslint-disable camelcase */
+const { SHA256 } = require("crypto-js");
+const get = require("get-value");
+const { EventType } = require("../../../constants");
+const {
+  getErrorRespEvents,
+  CustomError,
+  constructPayload,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultBatchRequestConfig,
+  getSuccessRespEvents,
+  isDefinedAndNotNullAndNotEmpty
+} = require("../../util");
+const {
+  trackMapping,
+  TRACK_ENDPOINT,
+  BATCH_ENDPOINT,
+  eventNameMapping,
+  MAX_BATCH_SIZE
+} = require("./config");
+
+function checkIfValidPhoneNumber(str) {
+  // Ref - https://ads.tiktok.com/marketing_api/docs?id=1701890979375106
+  // Regular expression to check whether it has country code
+  // but should not include +86
+  const regexExp = /^(\+(?!86)\d{1,3}){0,1}[0-9]{10}$/gi;
+
+  return regexExp.test(str);
+}
+
+const trackResponseBuilder = async (message, { Config }) => {
+  const pixel_code = Config.pixelCode;
+
+  let event = get(message, "event");
+  event = event ? event.trim().toLowerCase() : event;
+  if (!event) {
+    throw new CustomError("Event name is required", 400);
+  }
+
+  if (eventNameMapping[event] === undefined) {
+    throw new CustomError(`Event name (${event}) is not valid`, 400);
+  }
+  event = eventNameMapping[event];
+
+  let payload = constructPayload(message, trackMapping);
+  payload = { pixel_code, event, ...payload };
+
+  /*
+   * Hashing user related detail i.e external_id, email, phone_number
+   */
+
+  if (Config.hashUserProperties) {
+    const external_id = get(payload, "context.user.external_id");
```

Comment:
Use our general `externalId` structure which resides in `context.externalId` and use our generic util method to extract it so that we maintain parity for rudder nomenclature

Reply:
Done